include/Authentication.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef AUTHENTICATION_HPP
+#define AUTHENTICATION_HPP
+
+#include <QObject>
+#include <QString>
+
+class User : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString username MEMBER username READ getUsername NOTIFY usernameChanged);
+    Q_PROPERTY(QString password MEMBER password READ getPassword NOTIFY passwordChanged);
+    Q_PROPERTY(bool isAuthenticated MEMBER authenticated READ isAuthenticated NOTIFY authenticatedChanged);
+
+public:
+    User(QObject *parent = nullptr);
+    User(const User &other);
```

Comment:
[Conditions for automatic generation of default/copy/move ctor and copy/move assignment operator](https://stackoverflow.com/questions/4943958/conditions-for-automatic-generation-of-default-copy-move-ctor-and-copy-move-assi)